plugins/vpc-eni/e2eTests/e2e_test.go

```diff
@@ -0,0 +1,258 @@
+//go:build e2e_test
+// +build e2e_test
+
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/aws/amazon-vpc-cni-plugins/network/eni"
+	"github.com/containernetworking/cni/pkg/invoke"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containernetworking/plugins/pkg/testutils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	testENIName1  = "vpc-eni-test-1" // expected to exist on the host
+	testENIName2  = "vpc-eni-test-2" // expected to exist on the host
+	ifName        = "eni-test-eth0"
+	containerID   = "contain-er"
+	netConfFormat = `
+{
+    "type":"vpc-eni",
+    "cniVersion":"1.0.0",
+    "name":"eni-test",
+    "eniName":"%s",
+    "eniMACAddress":"%s",
+    "eniIPAddresses":["%s", "%s"],
+    "gatewayIPAddresses":["%s"],
+    "useExistingNetwork":false,
+    "blockInstanceMetadata":true,
```

Comment:
I can see here blockInstanceMetadata=true, however later I see that in a validation we found the IMDS route. Do we need extra validation there to make sure IMDS is blocked? For example: https://github.com/aws/amazon-vpc-cni-plugins/blob/master/plugins/vpc-branch-eni/e2eTests/e2e_test.go#L232 . Please let me know if I am misunderstanding

Also should we add another add with blockInstanceMetadata=false?

Reply:
Thanks @ubhattacharjya. IMDS is blocked with a blackhole route, so yes a route for IMDS endpoint is expected when `blockInstanceMetadata=true`. 
https://github.com/aws/amazon-vpc-cni-plugins/blob/58a0e2ba13522ca2a3792a824a196c10babea3f4/network/imds/imds_linux.go#L37-L40

I think we should also assert that the route for IMDS endpoint is indeed a blackhole route if `blockInstanceMetadata=true`. I also agree that another test case should be added for `blockInstanceMetadata=false` case. I will make these changes! :)

Reply:
Made the updates :)